fin-advisor-crm/markets/views.py

```diff
@@ -1,47 +1,239 @@
 from django.shortcuts import render
-from django.http import JsonResponse
+from django.http import JsonResponse, HttpRequest
+from django.db.models import Q
+from django.views.decorators.csrf import csrf_exempt
 
 from .models import DailyPublicEquityTrades
 
-from datetime import datetime
+import json
+import datetime
+import requests
+import locale
 
 
-# Create your views here.
-def get_summary_card_info():
+import os
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_ollama.llms import OllamaLLM
+
+
+locale.setlocale(locale.LC_ALL, 'en_US.UTF-8')
+
+
+EQUITY_DB_DATE_COL = "trade_date"
+EQUITY_DB_PRICE_COL = "closing_price"
+
+
+def format_currency(value):
+    if value >= 1_000_000:
+        return "GHS{:,.0f} M".format(value / 1_000_000)
+    elif value >= 1_000:
+        return "GHS{:,.0f} K".format(value / 1_000)
+    else:
+        return "GHS{:,.0f}".format(value)
```

Comment:
Best to put this in `core/utils.py` so its available for use throughout the application.